AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift

```diff
@@ -940,8 +940,18 @@ extension AWSMobileClient {
             completionHandler(nil, AWSMobileClientError.notSignedIn(message: notSignedInErrorMessage))
             return
         }
-        let userDetails = AWSMobileClientUserDetails(with: self.userpoolOpsHelper.currentActiveUser!)
-        userDetails.getUserAttributes(completionHandler: completionHandler)
+        self.getTokens { _, error in
+            guard error == nil else {
+                completionHandler(nil, error)
+                return
+            }
+            guard let currentActiveUser = self.userpoolOpsHelper.currentActiveUser else {
```

Comment:
Can we move this outside the self.getTokens? So that we will fail faster if the user is not signed in.